quimb/tensor/tensor_dmrg.py

```diff
@@ -866,24 +866,39 @@ def sweep(self, direction, canonize=True, verbosity=0, **update_opts):
             Sweep from left to right (->) or right to left (<-) respectively.
         canonize : bool, optional
             Canonize the state first, not needed if doing alternate sweeps.
+        bounds : tuple, optional
+            A 2-tuple with cites (from and to, not included) to perform the sweep
+            between.
         verbosity : {0, 1, 2}, optional
             Show a progress bar for the sweep.
         update_opts :
             Supplied to ``self._update_local_state``.
         """
-        if canonize:
-            {'R': self._k.right_canonize,
-             'L': self._k.left_canonize}[direction](bra=self._b)
 
         n, bsz = self.n, self.bsz
 
-        direction, begin, sweep = {
-            ('R', False): ('right', 'left', range(0, n - bsz + 1)),
-            ('L', False): ('left', 'right', range(n - bsz, -1, -1)),
-            ('R', True): ('right', 'left', range(0, n)),
-            ('L', True): ('left', 'right', range(n - 1, -1, -1)),
+        direction, begin, bounds_ext = {
+            ('R', False): ('right', 'left', (0, n - bsz + 1)),
+            ('L', False): ('left', 'right', (n - bsz, -1)),
+            ('R', True): ('right', 'left', (0, n)),
+            ('L', True): ('left', 'right', (n - 1, -1)),
         }[direction, self.cyclic]
 
+        if bounds is None:
+            bounds = bounds_ext
+
+        if (direction == "right" and bounds[0] >= bounds[1]) or (direction == "left" and bounds[0] <= bounds[1]):
+            raise ValueError(f"Invalid bounds for direction={direction} (wrong order): {bounds}")
```

Comment:
Although f-strings are really nice, it would be good to stick to ``.format`` syntax for the moment, since notionally ``quimb`` should still support python 3.5.